src/ciadmin/generate/tcyml.py

```diff
@@ -49,15 +49,13 @@ async def get(repo_path, repo_type="hg", revision=None, default_branch=None):
         if repo_path in _cache:
             return _cache[repo_path]
 
-        try:
-            async with aiohttp_session().get(url) as response:
-                response.raise_for_status()
-                result = await response.read()
-        except aiohttp.ClientResponseError as e:
-            if e.status == 404:
-                result = None
-            else:
-                raise e
+        client = RetryClient(
+            client_session=aiohttp_session(),
+            retry_options=ExponentialRetry(attempts=5, statuses={404}),
```

Comment:
It seems odd to me that a 404 would be intermittent; are we sure that's what's happening?

We should probably retry on 5xx though, if we're adding retries...